x-pack/plugins/infra/server/routes/metrics_explorer/lib/get_dataset_for_field.ts

```diff
@@ -17,15 +17,31 @@ interface EventDatasetHit {
 export const getDatasetForField = async (
   client: ESSearchClient,
   field: string,
-  indexPattern: string
+  indexPattern: string,
+  timerange: { field: string; to: number; from: number }
 ) => {
   const params = {
     allowNoIndices: true,
     ignoreUnavailable: true,
     terminateAfter: 1,
     index: indexPattern,
     body: {
-      query: { exists: { field } },
+      query: {
+        bool: {
+          filter: [
+            { exists: { field } },
+            {
+              range: {
+                [timerange.field]: {
+                  gte: timerange.from,
+                  lte: timerange.to,
+                  format: 'epoch_millis',
+                },
+              },
+            },
```

Comment:
Will this query change make the experience different for some cases now?

Reply:
It shouldn't, actually this should make it more accurate since we are limiting it to the same data that's being visualized.